core/commonMain/src/implementations/immutableMap/PersistentHashMapBuilder.kt

```diff
@@ -61,6 +62,18 @@ internal class PersistentHashMapBuilder<K, V>(private var map: PersistentHashMap
         return operationResult
     }
 
+    override fun putAll(from: Map<out K, V>) {
+        val map = from as? PersistentHashMap ?: (from as? PersistentHashMapBuilder)?.map
```

Comment:
Hmm, this looks suspicious: `(from as? PersistentHashMapBuilder)?.map`. This returns the original immutable map from which builder was created. When we later get the root `node` of that map, it may be different from the root node of the builder if it has been already modified.
Please add a test for that.